From a003dfd93caf7d31a189f59f27bbc8bbe3864a88 Mon Sep 17 00:00:00 2001
From: Oliver Hartkopp <oliver@hartkopp.net>
Date: Thu, 27 Dec 2007 16:50:06 -0800
Subject: [PATCH] [CAN]: Fix plain integer definitions in userspace header.

This patch fixes the use of plain integers instead of __u32 in a struct
that is visible from kernel space and user space.

Thanks to Sam Ravnborg for pointing out the wrong plain int usage.

Signed-off-by: Oliver Hartkopp <oliver@hartkopp.net>
Acked-by: Sam Ravnborg <sam@ravnborg.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 include/linux/can/bcm.h |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/can/bcm.h b/include/linux/can/bcm.h
index 7ade33a..7f29327 100644
--- a/include/linux/can/bcm.h
+++ b/include/linux/can/bcm.h
@@ -26,12 +26,12 @@
  * @frames:    array of CAN frames.
  */
 struct bcm_msg_head {
-	int opcode;
-	int flags;
-	int count;
+	__u32 opcode;
+	__u32 flags;
+	__u32 count;
 	struct timeval ival1, ival2;
 	canid_t can_id;
-	int nframes;
+	__u32 nframes;
 	struct can_frame frames[0];
 };
 
-- 
1.5.4.4

